plugin/trino-teradata/src/test/java/io/trino/plugin/teradata/TestTeradataDatabase.java

```diff
@@ -22,37 +25,77 @@
 public class TestTeradataDatabase
         implements AutoCloseable, SqlExecutor
 {
+    private final ClearScapeManager clearScapeManager;
+    private final boolean useClearScape;
     private final String databaseName;
-    private final Connection connection;
+    private Connection connection;
     private final String jdbcUrl;
     private final Map<String, String> connectionProperties = new HashMap<>();
 
-    /**
-     * Creates a new TestTeradataDatabase instance using the provided configuration.
-     *
-     * @param config DatabaseConfig containing connection details.
-     * @throws SQLException if a database access error occurs.
-     * @throws ClassNotFoundException if the JDBC driver class is not found.
-     */
-    public TestTeradataDatabase(DatabaseConfig config)
+    private void initializeConnection(DatabaseConfig config)
     {
-        this.databaseName = config.getDatabaseName();
-        this.jdbcUrl = config.getJdbcUrl();
+        String username;
+        String password;
+
+        // Use ClearScape credentials if available, otherwise fall back to config
+        if (useClearScape && clearScapeManager != null) {
+            JsonNode authInfo = clearScapeManager.getConfigJSON().get(TeradataConstants.LOG_MECH);
+            username = authInfo.get("username").asText();
+            password = authInfo.get("password").asText();
+            System.out.println("Using ClearScape credentials");
+        }
+        else {
+            username = config.getUsername();
+            password = config.getPassword();
+            System.out.println("Using environment variable credentials");
+        }
 
         connectionProperties.put("connection-url", jdbcUrl);
-        connectionProperties.put("connection-user", config.getUsername());
-        connectionProperties.put("connection-password", config.getPassword());
+        connectionProperties.put("connection-user", username);
+        connectionProperties.put("connection-password", password);
         connectionProperties.put("join-pushdown.enabled", "true");
-        //connectionProperties.put("join-pushdown.strategy", "EAGER");
+
         try {
-            Class.forName("com.teradata.jdbc.TeraDriver");
-            this.connection = DriverManager.getConnection(jdbcUrl, config.getUsername(), config.getPassword());
+            Class.forName(TeradataConstants.DRIVER_CLASS);
+            this.connection = DriverManager.getConnection(jdbcUrl, username, password);
         }
         catch (SQLException | ClassNotFoundException e) {
-            throw new RuntimeException(e);
+            throw new RuntimeException("Failed to initialize database connection", e);
+        }
+    }
+
+    public TestTeradataDatabase(DatabaseConfig config, boolean useClearScape)
```

Comment:
No need extra parameter for useClearScape. Remove it as config has already a property for this